Fix #2942: Document supported platforms (#3065)
mozilla/DeepSpeech: doc/SUPPORTED_PLATFORMS.rst

.. _supported-platforms-inference:

Supported platforms for inference
=================================

Here we maintain the list of supported platforms for running inference.

Linux / AMD64 without GPU
^^^^^^^^^^^^^^^^^^^^^^^^^

> * x86-64 CPU with AVX/FMA (one can rebuild without, but it might slow processing)

Reviewer: ```suggestion
* x86-64 CPU with AVX/FMA (one can rebuild without AVX/FMA, but it might slow down inference)
```

* Ubuntu 14.04+ (glibc >= 2.19, libstdc++6 >= 4.8)

> * Full TensorFlow runtime
> * TensorFlow Lite runtime

Reviewer: ```suggestion
* Full TensorFlow runtime (``deepspeech`` packages)
* TensorFlow Lite runtime (``deepspeech-tflite`` packages)
```

Linux / AMD64 with GPU
^^^^^^^^^^^^^^^^^^^^^^

> * x86-64 CPU with AVX/FMA (one can rebuild without, but it might slow processing)

Reviewer: ```suggestion
* x86-64 CPU with AVX/FMA (one can rebuild without AVX/FMA, but it might slow down inference)
```

* Ubuntu 14.04+ (glibc >= 2.19, libstdc++6 >= 4.8)
* CUDA 10.0 (and capable GPU)

> * Full TensorFlow runtime
> * TensorFlow Lite runtime

Reviewer: ```suggestion
* Full TensorFlow runtime (``deepspeech-gpu`` packages)
* TensorFlow Lite runtime (``deepspeech-tflite`` packages)
```

Linux / ARMv7
^^^^^^^^^^^^^

> * Cortex-A53 compatible SoC ARMv7 + NEON

Reviewer: ```suggestion
* Cortex-A53 compatible ARMv7 SoC with Neon support
```

* Raspbian Buster-compatible distribution

> * TensorFlow lite runtime

Reviewer: ```suggestion
* TensorFlow Lite runtime
```

Linux / Aarch64
^^^^^^^^^^^^^^^

> * Cortex-A72 compatible SoC Aarch64

Reviewer: ```suggestion
* Cortex-A72 compatible Aarch64 SoC
```

* ARMbian Buster-compatible distribution

> * TensorFlow lite runtime

Reviewer: ```suggestion
* TensorFlow Lite runtime
```

Android / ARMv7
^^^^^^^^^^^^^^^

> * SoC ARMv7 + NEON

Reviewer: ```suggestion
* ARMv7 SoC with Neon support
```

* Android 7.0-10.0
* NDK API level >= 21

> * TensorFlow lite runtime

Reviewer: ```suggestion
* TensorFlow Lite runtime
```

Android / Aarch64
^^^^^^^^^^^^^^^^^

> * SoC Aarch64

Reviewer: ```suggestion
* Aarch64 SoC
```

* Android 7.0-10.0
* NDK API level >= 21

> * TensorFlow lite runtime

Reviewer: ```suggestion
* TensorFlow Lite runtime
```

macOS / AMD64
^^^^^^^^^^^^^

> * x86-64 CPU with AVX/FMA (one can rebuild without, but it might slow processing)

Reviewer: ```suggestion
* x86-64 CPU with AVX/FMA (one can rebuild without AVX/FMA, but it might slow down inference)
```

* macOS >= 10.10

> * Full TensorFlow runtime
> * TensorFlow lite runtime

Reviewer: ```suggestion
* Full TensorFlow runtime (``deepspeech`` packages)
* TensorFlow Lite runtime (``deepspeech-tflite`` packages)
```

> Windows / AMD64

Reviewer: On the Linux section you separated without GPU and with GPU. Why not here too?

Author: because I forgot we had GPU support on Windows :)

^^^^^^^^^^^^^^^

> * x86-64 CPU with AVX/FMA (one can rebuild without, but it might slow processing)

Reviewer: ```suggestion
* x86-64 CPU with AVX/FMA (one can rebuild without AVX/FMA, but it might slow down inference)
```

* Windows Server >= 2012 R2 ; Windows >= 8.1

> * Full TensorFlow runtime
> * TensorFlow lite runtime

Reviewer: ```suggestion
* Full TensorFlow runtime (``deepspeech`` packages)
* TensorFlow Lite runtime (``deepspeech-tflite`` packages)
```
